tests/Analyzers/AnalyzerAssemblyGenerator.cs

```diff
@@ -19,6 +19,25 @@ namespace Microsoft.CodeAnalysis.Tools.Tests.Analyzers
 {
     public static class AnalyzerAssemblyGenerator
     {
+        private static readonly string s_basePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+        private static async Task<IEnumerable<MetadataReference>> GetReferencesAsync()
+        {
+            var references = new List<MetadataReference>()
+            {
+                MetadataReference.CreateFromFile(Path.Combine(s_basePath, Path.GetFileName(typeof(ImmutableArray).Assembly.Location))),
+                MetadataReference.CreateFromFile(Path.Combine(s_basePath, Path.GetFileName(typeof(SharedAttribute).Assembly.Location))),
+                MetadataReference.CreateFromFile(Path.Combine(s_basePath, Path.GetFileName(typeof(CSharpCompilation).Assembly.Location))),
+                MetadataReference.CreateFromFile(Path.Combine(s_basePath, Path.GetFileName(typeof(DiagnosticAnalyzer).Assembly.Location))),
+                MetadataReference.CreateFromFile(Path.Combine(s_basePath, Path.GetFileName(typeof(CodeFixProvider).Assembly.Location))),
+            };
+
+            var netcoreMetadataReferences = await ReferenceAssemblies.NetCore.NetCoreApp31.ResolveAsync(LanguageNames.CSharp, CancellationToken.None);
```

Comment:
Using the NetStandard2.0 references causes a version conflict with the System.Runtime version.